### PR TITLE
Feature/general improvement v1

### PR DESCRIPTION
**Pull request by BartoliniAndBorderCollies**
#### Issues
https://trello.com/c/x61tNbKG
https://trello.com/c/df0SRb1C
https://trello.com/c/S2M5oXTS

### Description
This branch improves code readability, reduces redundancy, and fixes some bugs.

### Motivation
In order to enhance certain aspects of the app, I created this branch and made some general improvements in areas where I saw opportunities.

### Changes introduced by this pull request
- rename of test methods for better clarity
- add Supplier to test cases (where before was null)
- add ResponseDTO class and use it in delete methods where before there was a ResponseEntity
- fix the bug with a report and null Supplier
- fix the bug InventoryItem with the same name
- refactor of magic numbers

### Testing
- [x] some tests were extended (giving a Supplier instead of null)
- [x] all Junit and integration tests are functional

### CODE REVIEW

File: src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java [hunk 1]
```diff
@@ -36,25 +36,27 @@ public InventoryItemDTOResponse create(InventoryItemDTORequest inventoryItemDTOR
...
 
         inventoryItem.setSupplier(supplier);
         inventoryItemRepository.save(inventoryItem);
 
         return modelMapper.map(inventoryItem, InventoryItemDTOResponse.class);
     }
 
-    private void checkIfThisInventoryItemWithThisSupplierAlreadyExists(InventoryItemDTORequest inventoryItemDTORequest) throws ObjectAlreadyExistException {
-        Optional<InventoryItem> optionalInventoryItem = inventoryItemRepository.findByName(inventoryItemDTORequest.getName());
+    private void checkIfThisInventoryItemWithThisSupplierAlreadyExists(InventoryItemDTORequest inventoryItemDTORequest)
```
Comment (JasinskiSz) [inline src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java]: Why such complicated name? It's very long and not descriptive.

**Comment by JasinskiSz** [comment]
According to PR description this branch introduces improvement of code readability, fixes some bugs and reduces redundancy. I'd actually appreciate to split these at least into two different PRs, where one would be for bug fixes only (even more preferably would be to have every PR for every bug fix if these are major issues) and another one for code readability improvements.

That's just something to keep in mind for future PRs and working with git. Also a better description of bugs in those separate PRs would be even more helpful and appreciated. In Trello those are not described enough, too.

Right now it's hard checking the code and easy to miss anything important.

File: src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java [hunk 2]
```diff
@@ -21,7 +22,7 @@ public InventoryItemDTOResponse(Long id, int amount, Supplier supplier,
 
     private Long id;
     private int amount;
-    private Supplier supplier; //TODO: change to DTO
```
Comment (JasinskiSz) [inline src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java:24]: Why it's not changed into DTO then? Conception changed here?

Comment (BartoliniAndBorderCollies) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java:24]: exactly! SupplierDTO is the same as Supplier so I decided to stay as it is

File: src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java (hunk 1 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java]: and you see for me it is exact and very descriptive - therefore it is so long -> this methods checks if such InventoryItem with specific Supplier exists that is why I added 'This' twice - to highlight it.
anyways
what I can propose instead?
maybe: checkInventoryItemWithSupplierExists() ?

**Comment by BartoliniAndBorderCollies** [comment]
I left some answers.
Shall I make a new PR or just keep in mind what you wrote and this one stays as it is?

**Comment by JasinskiSz** [comment]
> I left some answers. Shall I make a new PR or just keep in mind what you wrote and this one stays as it is?

Leave it as is. Just keep in mind for the next time :)

File: src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java (hunk 2 above)
Comment (JasinskiSz) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java:24]: Okay, I see. I'd add a DTO anyway, so we don't expose our model to the outside world. That's the first point.

Second would be about expadability of the project. You decide if this solution satisfies you atm.

File: src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java (hunk 1 above)
Comment (JasinskiSz) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java]: Look, @BartoliniAndBorderCollies, long names doesn't improve readability. You have comments available to describe the code. Especially, when you can't think of a better name for a method.

I'd suggest using:
`assertUniqueInventorySupplier`
`verifyItemWithSupplierExists`
`doesItemWithSupplierExist`
`confirmItemSupplierUniqueness`

or a more custom ones:
`validateItemSupplierCombination`
`checkForDuplicateItemSupplierPair`
`ensureItemAndSupplierUniqueness`

Your case here is so specific that's the best way to approach is to get method a proper documentation and a simple name.

**Review by JasinskiSz — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java (hunk 1 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java]: got it!

File: src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java (hunk 2 above)
Comment (BartoliniAndBorderCollies) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java:24]: this little change in code would bring dozens of changes. I tried to do this and this is like domino. I don't feel this is worth to do right now. It demands a lot of work and I don't see an add value at this moment. Furthermore this could also lead to malfunction of the app.

Comment (BartoliniAndBorderCollies) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java:24]: I'm refactoring right now!

Comment (BartoliniAndBorderCollies) [reply src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/InventoryItemDTOs/InventoryItemDTOResponse.java:24]: okay! after the struggle I think it is much better now!

**Review by JasinskiSz — APPROVED** [review]
